apis/gasbuddy.js

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file apis/gasbuddy.js
+ *
+ * @author fewieden
+ * @license MIT
+ *
+ * @see  https://github.com/fewieden/MMM-Fuel
+ */
+
+/**
+ * @external node-fetch
+ * @see https://www.npmjs.com/package/node-fetch
+ */
+const fetch = require('node-fetch');
+
+/**
+ * @external node-html-parser
+ * @see https://www.npmjs.com/package/node-html-parser
+ */
+const { parse } = require('node-html-parser');
+
+const BASE_URL = 'https://www.gasbuddy.com';
+const MAX_PAGE = 1;
+
+let config;
+
+/**
+ * @function getRequestPaths
+ * @description URL paths for fuel type to request data sorted by distance and by price.
+ *
+ * @param {string} type - Fuel type.
+ *
+ * @returns {string[]} URL paths for fuel type.
+ */
+function getRequestPaths(type) {
+    switch(type) {
+        case 'midgrade':
+          type = '2';
+          break;
+        case 'premium':
+          type = '3';
+          break;
+        case 'diesel':
+           type = '4';
+           break;
+        case 'e85':
+           type = '5';
+           break;
+        case 'unl88':
+            type = '12';
+            break;
+        default:
+          type = '1';
+      }
+    return [
+        `/home?search=${config.zip}&fuel=${type}&maxAge=0&method=all`
```

Comment:
No need to return an array here if it is always 1 element. This will simplify also other parts as you can get rid of loops.